File: neetcode/problems/leetcode75/solution.js
const sortColors = nums => {
	const bucket = [0, 0, 0];

	for (const num of nums) {
		bucket[num]++;
	}

	let idx = 0;

	for (let i = 0; i < bucket.length; i++) {
		for (let j = 0; j < bucket[i]; j++) {
			nums[idx] = i;
			idx++;
		}
	}

	return nums;
};

File: neetcode/problems/leetcode75/solution.py
from typing import List


class Solution:
    def sortColors(self, nums: List[int]) -> None:
        bucket = [0, 0, 0]
        idx = 0

        for num in nums:
            bucket[num] += 1

        for i in range(3):
            for j in range(bucket[i]):
                nums[idx] = i
                idx += 1
